packages/number-field/src/NumberField.ts

```diff
@@ -496,6 +496,18 @@ export class NumberField extends TextfieldBase {
 
     protected override handleInput(event: InputEvent): void {
         if (this.isComposing) {
+            // If user actually types a new character.
+            if (event.data) {
```

Comment:
If event.data is null, it could lead to unexpected behavior. For example, pressing backspace or delete can cause event.data to be empty.

Reply:
An early return would be more tactical!
```
if (event.data === null || event.data === '') {
    return;
}
```

Reply:
`if (event.data)` also guards for `null` and empty string, and after this this block is a `return` at line 511.
Wouldn't it work the same?